src/main/java/seedu/address/logic/commands/FindCommand.java

```diff
@@ -16,15 +21,33 @@
             + "Example: " + COMMAND_WORD + " alice bob charlie";
 
     private final Set<String> keywords;
+    private final Deadline deadline;
+
+    public FindCommand(String keywords, String deadline)
+            throws IllegalValueException {
+        final Set<String> keywordSet = new HashSet<>();
+        StringTokenizer st = new StringTokenizer(keywords, " ");
+        while (st.hasMoreTokens()) {
+            keywordSet.add(st.nextToken());
+        }
+        this.keywords = keywordSet;
+        this.deadline = createDeadline(deadline);
 
-    public FindCommand(Set<String> keywords) {
-        this.keywords = keywords;
     }
 
     @Override
     public CommandResult execute() {
-        model.updateFilteredTaskList(keywords);
-        return new CommandResult(getMessageForTaskListShownSummary(model.getFilteredTaskList().size()));
+        if (!keywords.equals("")) {
```

Comment:
`keywords` is a Set of strings so it will never equal `""` here. Also no need to have this if. My suggestion:
```
model.updateFilteredTaskListByName(keywords);
if (deadline != null) model.updateFilterTaskListByDate(deadline);
```